### PR TITLE
Add new device: Elecrow 5.0inch esp32 display

### PR DESCRIPTION
**Pull request by bruxy70**
5-inch 800*480 resolution LCD display with ESP32 S3 microcontroller

Including the minimal configuration.
Plus a more extensive example provided by Elecrow technician for the video I am making (don't want to keep it for myself). 
With the reference to Elecrow wiki with all parameters.

### CODE REVIEW

**Review by jesserockz — CHANGES_REQUESTED** [review]
I just tried this out and it doesn't work. :see_no_evil: 

- The lvgl image configurations are invalid as they do not have a `src`.
- Dont use the `esp32s3box`, use the generic `esp32-s3-devkitc-1`


Example configurations should not have secrets and passwords and should not make any assumptions about data that may or may not be available.

It is fine to have an lvgl example in here, but maybe it could be a bit more useful with buttons etc to show the touch working.

You should not add extra "soft" sensors like uptime, wifi info etc. Only hardware sensors present on a board should be included.

If you want a `1s` update interval for the counter, use the `interval` component.

**Comment by bruxy70** [comment]
Hi. Thanks for the feedback. For lvgl part I did include the official example I got from the manufacturer. But I agree with you that it was too complicated and partially incorrect, so I made own example with one button and one meter.
I was thinking whether to include the lvgl snippet in general, but I think this is realistically much closer to the real use-case that the generic set of ESPHome graphics functions, and lvgl works much better on the display anyway.

**Comment by sprior** [comment]
I have been following this merge because I'm using a slight variation of the Elecrow display.  I was just able to get the LVGL version of the example code working with one exception - the 3 power_meter_input_img's are not defined in the example.  I got it working by commenting out the "- image:" sections of each but it would be nice if this part of the example was fixed so we could see what it's supposed to look like.

**Comment by bruxy70** [comment]
@sprior Sorry, I do not follow. I did change the lvgl example based on @jesserockz feedback 2 days ago - it does not contain the 3 images you mention.
There is now one and that one is defined. I also removed the API to HA. And added a button. 
Or if I do not see it, what line number do you refer to?

**Comment by sprior** [comment]
@bruxy70 Ok thanks, I was looking at the version on the website, didn't see the fixed version.  I'm working with the MaTouch ESP32-S3 Parallel TFT with Touch 7“ (1024x600) board which so far I've got working with:
`display:
  - platform: rpi_dpi_rgb
    id: main_display
    color_order: RGB
    invert_colors: True
    update_interval: never
    auto_clear_enabled: false # takes too long to clear the display
    dimensions:
      width: 1024
      height: 600
    de_pin: 40
    hsync_pin: 39
    vsync_pin: 41
    pclk_pin: 42
    pclk_frequency: 16MHz
    #
    hsync_front_porch: 40
    hsync_pulse_width: 48
    hsync_back_porch: 28
    vsync_front_porch: 13
    vsync_pulse_width: 3
    vsync_back_porch: 45
    #
    data_pins:
      red:
        - 45        #r1
        - 48        #r2
        - 47        #r3
        - 21        #r4
        - 14        #r5
      green:
        - 5         #g0
        - 6         #g1
        - 7         #g2
        - 15        #g3
        - 16        #g4
        - 4         #g5
      blue:
        - 8         #b1
        - 3         #b2
        - 46        #b3
        - 9         #b4
        - 1         #b5`

And uses different pins for I2C:
`
i2c:
  sda: GPIO17
  scl: GPIO18
  scan: true`

**Comment by sprior** [comment]
@bruxy70 I just noticed that in the latest version of your index.md file you still mention esp32s3box  up in the top Drivers section.

**Comment by bruxy70** [comment]
> @bruxy70 I just noticed that in the latest version of your index.md file you still mention esp32s3box up in the top Drivers section.

Thanks. I missed that one. Fixed. The examples are tested and working, so I hope it should now be ready to go?

**Comment by bruxy70** [comment]
The status says change requested.  But I made the changes. What do I have to update to update the status?

**Comment by jesserockz** [comment]
> The status says change requested. But I made the changes. What do I have to update to update the status?

Nothing. Someone (if not me) will come back around when time permits =)

File: src/docs/devices/Elecrow-5inch-ESP32-display/index.md [hunk 1]
```diff
@@ -0,0 +1,402 @@
...
+        - 7         #g2
+        - 15        #g3
+        - 16        #g4
+        - 4         #g5
+      blue:
+        - 8         #b1
+        - 3         #b2
+        - 46        #b3
+        - 9         #b4
+        - 1         #b5
```
Comment (jesserockz) [inline src/docs/devices/Elecrow-5inch-ESP32-display/index.md]:
```suggestion
red:
  - 45        #r1
  - 48        #r2
  - 47        #r3
  - 21        #r4
  - 14        #r5
green:
  - 5         #g0
  - 6         #g1
  - 7         #g2
  - 15        #g3
  - 16        #g4
  - 4         #g5
blue:
  - 8         #b1
  - 3         #b2
  - 46        #b3
  - 9         #b4
  - 1         #b5
```

File: src/docs/devices/Elecrow-5inch-ESP32-display/index.md [hunk 2]
```diff
@@ -0,0 +1,402 @@
...
+  - 1         #b5
+```
+
+## Simle configuration
+
+```yaml
+esphome:
+  name: display
+  platformio_options:
+    build_flags: "-DBOARD_HAS_PSRAM"
```
Comment (jesserockz) [inline src/docs/devices/Elecrow-5inch-ESP32-display/index.md]: Not required as this is set automatically with the `psram` component when using Arduino
```suggestion
```

File: src/docs/devices/Elecrow-5inch-ESP32-display/index.md [hunk 3]
```diff
@@ -0,0 +1,401 @@
...
+## Example (using LGVL graphics library)
+
+```yaml
+substitutions:
+  device_name: display
+
+esphome:
+  name: ${device_name}
+  platformio_options:
+    build_flags: "-DBOARD_HAS_PSRAM"
```
Comment (jesserockz) [inline src/docs/devices/Elecrow-5inch-ESP32-display/index.md]:
```suggestion
```

**Review by jesserockz — APPROVED** [review]
Thanks for making the changes/simplifications.
Compiled and installed this yaml as is and working nicely.

File: src/docs/devices/Elecrow-5inch-ESP32-display/index.md [hunk 4]
```diff
@@ -0,0 +1,400 @@
+---
+title: CrowPanel 5.0" HMI ESP32 display (by Elecrow)
+date-published: 2024-08-17
```
Comment (jesserockz) [inline src/docs/devices/Elecrow-5inch-ESP32-display/index.md]:
```suggestion
date-published: 2024-09-03
```

**Review by jesserockz — APPROVED** [review]

File: src/docs/devices/Elecrow-5inch-ESP32-display/index.md [hunk 5]
```diff
@@ -0,0 +1,400 @@
...
+
+## Drivers
+
+* Processor: ESP32-S3-WROOM-1-N4R8 (using `esp32-s3-devkitc-1` board)
+* Touchscreen: `gt911`
+* Display: ILI6122 & ILI5960 (using `rpi_dpi_rgb` plaform)
+
+## GPIO Pinout
+
+### i2c (used for touchscreen)
```
Comment (jesserockz) [inline src/docs/devices/Elecrow-5inch-ESP32-display/index.md]:
```suggestion
### I²C (used for touchscreen)
```

File: src/docs/devices/Elecrow-5inch-ESP32-display/index.md [hunk 6]
```diff
@@ -0,0 +1,400 @@
...
+
+| Pin    | Function      |
+| ------ | ------------- |
+| GPIO19 | i2c SDA     |
+| GPIO20 | i2c SCL     |
+
+### backlight
```
Comment (jesserockz) [inline src/docs/devices/Elecrow-5inch-ESP32-display/index.md]:
```suggestion
### Backlight
```

File: src/docs/devices/Elecrow-5inch-ESP32-display/index.md [hunk 7]
```diff
@@ -0,0 +1,400 @@
...
+
+| Pin    | Function      |
+| ------ | ------------- |
+| GPIO2  | backlight   |
+
+### display
```
Comment (jesserockz) [inline src/docs/devices/Elecrow-5inch-ESP32-display/index.md]:
```suggestion
### Display
```

File: src/docs/devices/Elecrow-5inch-ESP32-display/index.md [hunk 8]
```diff
@@ -0,0 +1,400 @@
...
+
+| Pin    | Function      |
+| ------ | ------------- |
+| GPIO40 | de_pin      |
+| GPIO39 | hsync_pin   |
+| GPIO41 | vsync_pin   |
+| GPIO0  | pclk_pin    |
+
+### data pins
```
Comment (jesserockz) [inline src/docs/devices/Elecrow-5inch-ESP32-display/index.md]:
```suggestion
### Display Data pins
```

File: src/docs/devices/Elecrow-5inch-ESP32-display/index.md [hunk 9]
```diff
@@ -0,0 +1,400 @@
...
+  - 4         #g5
+blue:
+  - 8         #b1
+  - 3         #b2
+  - 46        #b3
+  - 9         #b4
+  - 1         #b5
+```
+
+## Simle configuration
```
Comment (jesserockz) [inline src/docs/devices/Elecrow-5inch-ESP32-display/index.md]:
```suggestion
## Simple Configuration
```

File: src/docs/devices/Elecrow-5inch-ESP32-display/index.md [hunk 10]
```diff
@@ -0,0 +1,400 @@
...
+    x_min: 10
+    x_max: 240
+    y_min: 10
+    y_max: 90
+    on_press:
+      - lambda: |-
+            ESP_LOGI("btn", "Button pressed");
+```
+
+## Example (using LGVL graphics library)
```
Comment (jesserockz) [inline src/docs/devices/Elecrow-5inch-ESP32-display/index.md]:
```suggestion
## Advanced Configuration (using LGVL graphics library)
```

**Review by jesserockz — COMMENTED** [review]
Minor capitalisation and spelling

**Pull request merged**